docs/aws/audit/redshiftmonitoring/rules/redshift_reserved_node_payment_failed_remediation.mdx

```diff
@@ -1,6 +1,229 @@
+   - Enable detailed cost and usage reports to track your spending and identify any anomalies or unexpected charges. This helps in managing your budget effectively and preventing payment issues.
+
+By following these steps, you can proactively manage your billing and payment settings to prevent Redshift Reserved Nodes from having a status of "Payment Failed."
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent Redshift Reserved Nodes from having a status of "Payment Failed" using AWS CLI, you can follow these steps:
+
+1. **Monitor Billing and Payment Status:**
+   Regularly check your AWS billing and payment status to ensure that your payment methods are up-to-date and there are no issues with your account.
+
+   ```sh
+   aws ce get-cost-and-usage --time-period Start=$(date -d '1 month ago' +%Y-%m-%d),End=$(date +%Y-%m-%d) --granularity MONTHLY --metrics "UnblendedCost"
+   ```
+
+2. **Set Up Billing Alerts:**
+   Create billing alerts to notify you when your charges exceed a certain threshold. This helps in ensuring that you are aware of any potential payment issues before they affect your reserved nodes.
+
+   ```sh
+   aws cloudwatch put-metric-alarm --alarm-name "BillingAlarm" --metric-name "EstimatedCharges" --namespace "AWS/Billing" --statistic "Maximum" --period 21600 --threshold 100 --comparison-operator "GreaterThanOrEqualToThreshold" --dimensions Name=Currency,Value=USD --evaluation-periods 1 --alarm-actions arn:aws:sns:us-east-1:123456789012:MyTopic
+   ```
+
+3. **Update Payment Methods:**
+   Ensure that your payment methods are valid and up-to-date. You can update your payment methods through the AWS Management Console, but you can also use the AWS CLI to list your current payment methods and ensure they are valid.
+
+   ```sh
+   aws organizations list-payment-methods
+   ```
+
+4. **Automate Payment Verification:**
+   Use a script to periodically verify the status of your reserved nodes and ensure that there are no payment issues. This can be done using a combination of AWS CLI commands and a simple script.
+
+   ```sh
+   aws redshift describe-reserved-nodes --query 'ReservedNodes[?State==`payment-failed`]'
+   ```
+
+By following these steps, you can proactively monitor and manage your AWS billing and payment status to prevent Redshift Reserved Nodes from having a "Payment Failed" status.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the issue of Redshift Reserved Nodes having a status of "Payment Failed" using Python scripts, you can follow these steps:
+
+1. **Set Up AWS SDK (Boto3) and Authentication:**
+   Ensure you have the AWS SDK for Python (Boto3) installed and properly configured with your AWS credentials.
+
+   ```bash
+   pip install boto3
+   ```
+
+   ```python
+   import boto3
+
+   # Initialize a session using Amazon Redshift
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>